cs-amazon/src/main/java/io/cloudslang/content/amazon/factory/helpers/InstanceUtils.java

```diff
@@ -219,15 +262,121 @@ private void setModifyInstanceAttributeEbsSpecs(Map<String, String> queryParamsM
     private Map<String, String> getRebootStartStopTerminateCommonQueryParamsMap(InputsWrapper wrapper) {
         Map<String, String> queryParamsMap = new HashMap<>();
         InputsUtil.setCommonQueryParamsMap(queryParamsMap, wrapper.getCommonInputs().getAction(), wrapper.getCommonInputs().getVersion());
+        setInstanceIdsQueryParams(wrapper, queryParamsMap);
+
+        return queryParamsMap;
+    }
+
+    private String getCurrentVolumeType(Map<String, String> queryParamsMap, int index) {
+        return queryParamsMap.get(BLOCK_DEVICE_MAPPING + DOT + (index + ONE) + DOT + EBS + VOLUME_TYPE);
+    }
+
+    private boolean getOptionalsSetterFlag(String[] validationArray, String[] toBeValidatedArray, String firstInput, String secondInput) {
+        if (toBeValidatedArray != null) {
+            InputsUtil.validateAgainstDifferentArraysLength(validationArray, toBeValidatedArray, firstInput, secondInput);
+            return Boolean.TRUE;
+        }
+        return Boolean.FALSE;
+    }
+
+    private String getFilterValue(String input) {
```

Comment:
there is a problem in the logic of this method, if the input is "i386" for example it will go to default without any validation otherwise if the input is "architecture" it will go into validation and throw an exception